debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -169,6 +169,12 @@ else if (actualColumn.jdbcType() == Types.TINYINT || actualColumn.jdbcType() ==
             // read it again to get correct scale
             return rs.getObject(fieldNo) == null ? null : rs.getInt(fieldNo);
         }
+        // DBZ-2673
+        else if (actualColumn.typeName().equals("CHAR") ||
```

Comment:
Is there a specific reason for using the type-name here rather than jdbc-type as in prior if conditions?

Reply:
@Naros I actually had it like that at first, trouble was that MySQL's `ENUM' and `SET` were getting mapped to a JDBC char type which ended up causing the issues you saw earlier.